datafusion/tests/sql/functions.rs

```diff
@@ -17,6 +17,41 @@
 
 use super::*;
 
+#[tokio::test]
+async fn factorial() -> Result<()> {
+    let schema = Arc::new(Schema::new(vec![Field::new("c1", DataType::Float64, true)]));
+
+    let data = RecordBatch::try_new(
+        schema.clone(),
+        vec![Arc::new(Float64Array::from(vec![
+            Some(4.0),
+            Some(0.0),
+            Some(1.5),
+            Some(-1.0),
+            Some(10000000000.0),
+        ]))],
+    )?;
+    let table = MemTable::try_new(schema, vec![vec![data]])?;
+
+    let mut ctx = ExecutionContext::new();
+    ctx.register_table("test", Arc::new(table))?;
+    let sql = "SELECT factorial(c1) FROM test";
```

Comment:
I also recommend adding a constant here

```suggestion
    let sql = "SELECT factorial(5), factorial(c1) FROM test";
```

As well as tests for what happens if you pass in a integer column